File: frontend/src/api/plugins.ts
import * as PluginsIpc from '../../bindings/voxesis/src/Communication/InterProcess/pluginipc'
import {envIsWails} from "./common";
import {Plugin} from "../../bindings/voxesis/src/Common/Entity"

export async function GetPluginList(): Promise<Plugin[] | null> {
    if (envIsWails) {
        return PluginsIpc.GetPluginList()
    } else {
        const res = await fetch("/api/plugins/GetPluginList", {
            method: "GET",
            headers: {
                "Content-Type": "application/json"
            }
        })

        return res.json()
    }
}

export async function LoadPlugins(): Promise<string | null> {
    if (envIsWails) {
        return PluginsIpc.LoadPlugins()
    } else {
        const res = await fetch("/api/plugins/LoadPlugins", {
            method: "POST",
            headers: {
                "Content-Type": "application/json"
            }
        })

        return res.json()
    }
}

export default {
    GetPluginList,
    LoadPlugins
}

File: src/Web/Routes/Api/plugins.go
package v_web_api

import (
	vwebcontroller "voxesis/src/Communication/InterHttp"

	"github.com/gin-gonic/gin"
)

func Plugins(group *gin.RouterGroup) {
	ctrl := &vwebcontroller.Plugins{}

	group.POST("/LoadPlugins", ctrl.LoadPlugins)
	group.GET("/GetPluginList", ctrl.GetPluginList)
}
